modules/javafx.graphics/src/main/java/javafx/stage/Stage.java

```diff
@@ -297,6 +302,76 @@ public Stage(@NamedArg(value="style", defaultValue="DECORATED") StageStyle style
         super.show();
     }
 
+    /**
+     * Shows this stage at the specified location and adjusts the position as needed to keep the stage
+     * visible on screen. If the stage is already showing, it is moved to the computed position instead.
+     * <p>
+     * The {@code anchor} identifies a point on the stage that should coincide with the requested screen
+     * coordinates {@code (anchorX, anchorY)}. The stage location is computed from the anchor and the
+     * stage size, and is then adjusted to keep the stage within the screen bounds.
+     * <p>
+     * After the stage is shown, its {@link #xProperty() X} and {@link #yProperty() Y} properties will
+     * reflect the adjusted position.
+     * <p>
+     * Calling this method is equivalent to calling
+     * {@code show(anchorX, anchorY, anchor, ClampPolicy.BOTH, Insets.EMPTY)}.
+     *
+     * @param anchorX the requested horizontal location of the anchor point on the screen
+     * @param anchorY the requested vertical location of the anchor point on the screen
+     * @param anchor the point on the stage that should coincide with {@code (anchorX, anchorY)} on the screen
+     * @throws NullPointerException if {@code anchor} is {@code null}
+     * @since 26
+     */
+    public final void show(double anchorX, double anchorY, AnchorPoint anchor) {
```

Comment:
would it make more sense to move these new methods to `Window`, so they can also work for the `Popup` and its hierarchy ?

Reply:
Yeah, this is unfortunate.
Can we do anything to make the positioning of Popups and its hierarchy easier?  Maybe add another `show()` with a `ClampPolicy` argument?

Use case: I want to open a PopupWindow below the owner node, but: if the popup cannot be placed there because the whole thing is too close to the bottom of the screen, I want the popup to be **above** the node, and not on top of it as it works currently.